### DISCUSSION TITLE
How to save data to different table having same schema during runtime

### DISCUSSION THREAD

**Discussion by sameerdube9** [serenity-is/Serenity#5715] [Help]
Can i modify the table name inside SaveHandler based on condition in serenity?

**Comment by edwardch** [comment]
I haven't tried to do that on the fly, but what you could also do is create a new class `NewTableRow.cs`, then in your Create method of `OldTableRepository` you can set up the new entity and do a custom save request.

```cs
public SaveResponse Create(IUnitOfWork uow, SaveRequest<MyRow> request)
{
    public SaveResponse Create(IUnitOfWork uow, SaveRequest<MyRow> request)
    {
        var newEntity = new NewTableRow
        {
            ColumnA = request.Entity.ColumnA,
            ColumnB = request.Entity.ColumnB
            // etc
        };

        var saveRequest = new SaveRequest<NewTableRow> {Entity = newEntity};
        
        return new SaveRequestHandler<NewTableRow>().Process(uow, saveRequest, SaveRequestType.Create);
    }
}
```

You can add some extra logic to determine whether you want to save the data in OldTable or NewTable accordingly.
